### PR TITLE
Azure create virtual machine with an Availability Set associate

### PR DESCRIPTION
**Pull request by yuwzho**
##### ISSUE TYPE

 - Feature Pull Request

##### ANSIBLE VERSION

```
2.4
```

##### ADDITIONAL INFORMATION

related to issue #26003 

```

```

### CODE REVIEW

File: test/integration/targets/azure_rm_virtualmachine/tasks/virtualmachine.yml [hunk 1]
```diff
@@ -16,6 +16,11 @@
       name: "{{ storage_account }}"
       account_type: Standard_LRS
 
+- name: Create an availability set
+  azure_rm_availabilityset:
+    name: myavailabilityset
```
Comment (yaweiw) [inline test/integration/targets/azure_rm_virtualmachine/tasks/virtualmachine.yml]: can we not use a hardcoded name here?

File: lib/ansible/modules/cloud/azure/azure_rm_virtualmachine.py [hunk 2]
```diff
@@ -607,6 +615,7 @@ def __init__(self):
             ssh_password_enabled=dict(type='bool', default=True),
             ssh_public_keys=dict(type='list'),
             image=dict(type='dict'),
+            availability_set=dict(type='str'),
```
Comment (yaweiw) [inline lib/ansible/modules/cloud/azure/azure_rm_virtualmachine.py]: add: default=None?

File: lib/ansible/modules/cloud/azure/azure_rm_virtualmachine.py [hunk 3]
```diff
@@ -96,6 +96,10 @@
               offer, sku and version. NOTE: set image.version to 'latest' to get the most recent version of a given
               image."
         required: true
+    availability_set:
+        description:
+            - Name or ID of an existing availability set to add the VM to. None by default.
```
Comment (yaweiw) [inline lib/ansible/modules/cloud/azure/azure_rm_virtualmachine.py]: add:
    default: null

**Review by yaweiw — CHANGES_REQUESTED** [review]
some minor changes required

File: lib/ansible/modules/cloud/azure/azure_rm_virtualmachine.py (hunk 3 above)
Comment (xscript) [reply lib/ansible/modules/cloud/azure/azure_rm_virtualmachine.py]: Also we should tell user that this availability set should be in the same resource group of current vm.

File: lib/ansible/modules/cloud/azure/azure_rm_virtualmachine.py (hunk 2 above)
Comment (yuwzho) [reply lib/ansible/modules/cloud/azure/azure_rm_virtualmachine.py]: Actually the default is None, so there is no need to add one

Comment (yuwzho) [reply lib/ansible/modules/cloud/azure/azure_rm_virtualmachine.py]: Actually the default is None, so there is no need to add one

File: lib/ansible/modules/cloud/azure/azure_rm_virtualmachine.py (hunk 3 above)
Comment (yuwzho) [reply lib/ansible/modules/cloud/azure/azure_rm_virtualmachine.py]: fixed

File: test/integration/targets/azure_rm_virtualmachine/tasks/virtualmachine.yml (hunk 1 above)
Comment (yuwzho) [reply test/integration/targets/azure_rm_virtualmachine/tasks/virtualmachine.yml]: fixed

**Comment by yaweiw** [comment]
shipit

**Comment by xscript** [comment]
shipit

**Comment by jborean93** [comment]
Hi @yuwzho are you able to rebase the PR as there has been a recent push to devel that is causing some conflicts. This page http://docs.ansible.com/ansible/latest/dev_guide/developing_rebasing.html has more details on how to do this.

**Comment by yuwzho** [comment]
@jborean93 I have rebased the code to devel branch now 😄

**Review by nitzmahone — APPROVED** [review]

**Comment by nitzmahone** [comment]
Thanks!

**Pull request merged**